File: numero_secreto/numero_secreto.c
#include <stdio.h>
#define NUMERO_SECRETO 42
#define TENTATIVAS 10
int main()
{
    int i, palpite;
    for(i = 1; i <= 10;i++)
    {
        printf("Tentativa %d de %d.\n", i,TENTATIVAS);
        printf("Digite um numero inteiro:\n");
        scanf("%d", &palpite);
        if(palpite < NUMERO_SECRETO)
            printf("Tente um numero maior!\n");
        else if(palpite > NUMERO_SECRETO)
            printf("Tente um numero menor!\n");
        else {
            printf("Voce Acertou!");
            break;
        }
    }
    if(i > TENTATIVAS)
        printf("Numeros de tentativas esgotadas!\nO numero secreto eh %d", NUMERO_SECRETO);
    return 0;
}

File: numero_secreto/numero_secreto.py
NUMERO_SECRETO = 42
TENTATIVAS = 10

for i in range(1, TENTATIVAS + 1):
    print(f"Tentativa {i} de {TENTATIVAS}.\n")
    palpite = int(input("Digite um número inteiro:\n"))

    if palpite < NUMERO_SECRETO:
        print("Tente um número maior!\n")
    elif palpite > NUMERO_SECRETO:
        print("Tente um número menor!\n")
    else:
        print("Você acertou!")
        break
if i > TENTATIVAS:
    print("Número de tentativas esgotadas!\nO número secreto é", NUMERO_SECRETO)